csi/controller_server.go

```diff
@@ -133,12 +133,16 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 
 	// Extract AccessibleTopology from AccessibilityRequirements
 	// This will be used to set PV nodeAffinity via external-provisioner
-	accessibleTopology := cs.getAccessibleTopologyFromRequirements(req.GetAccessibilityRequirements())
 
-	// Warn when strict-local dataLocality is combined with accessibleTopology (PV nodeAffinity becomes immutable).
-	// Ref: https://github.com/longhorn/longhorn/issues/12261
-	if accessibleTopology != nil && volumeParameters["dataLocality"] == string(longhorn.DataLocalityStrictLocal) {
-		log.Warnf("CreateVolume for %s uses strict-local dataLocality with accessibleTopology; PV nodeAffinity will be immutable and may conflict with Longhorn strict-local behavior", volumeID)
+	var accessibleTopology []*csi.Topology
+
+	if volumeParameters["dataLocality"] == string(longhorn.DataLocalityStrictLocal) {
+		log.Infof(
+			"CreateVolume for %s ignores AccessibilityRequirements when dataLocality is strict-local since the volume will be created on the requested node only",
```

Comment:
```suggestion
			"CreateVolume for %s ignores AccessibilityRequirements because dataLocality is strict-local; the volume will be created only on the requested node",
```